src/services/new/entry/copy-entries-to-workbook.ts

```
@@ -0,0 +1,159 @@
+import {WorkbookPermission} from '../../../entities/workbook';
+import {copyToWorkbook} from '../../entry/actions';
+import {JoinedEntryRevisionColumns} from '../../../db/presentations';
+
+export type CopyEntriesToWorkbookParams = {
+    entryIds: string[];
+    workbookId: string;
+};
+
+const validateArgs = makeSchemaValidator({
+    type: 'object',
+    required: ['entryIds', 'workbookId'],
+    properties: {
+        entryIds: {
+            type: 'array',
+            items: {
+                type: 'string',
+            },
+        },
+        workbookId: {
+            type: 'string',
+        },
+    },
+});
+
+export const copyEntriesToWorkbook = async (
+    {ctx, trx, skipValidation = false, skipCheckPermissions = false}: ServiceArgs,
+    args: CopyEntriesToWorkbookParams,
+) => {
+    if (!skipValidation) {
+        validateArgs(args);
+    }
+
+    const {entryIds, workbookId: targetWorkbookId} = args;
+    const {tenantId, user} = ctx.get('info');
+    const updatedBy = makeUserId(user.userId);
+
+    logInfo(ctx, 'COPY_ENTRIES_TO_WORKBOOK_START', {
+        entryIds: entryIds.map((entryId) => Utils.encodeId(entryId)),
+        workbookId: Utils.encodeId(targetWorkbookId),
+        copiedBy: updatedBy,
+    });
+
+    if (entryIds.length === 0) {
+        throw new AppError(US_ERRORS.VALIDATION_ERROR, {
+            code: US_ERRORS.VALIDATION_ERROR,
+        });
+    }
+
+    const entries = await Entry.query(Entry.replica)
+        .select()
+        .whereIn(EntryColumn.EntryId, entryIds)
+        .andWhere({
+            [EntryColumn.TenantId]: tenantId,
+            [EntryColumn.IsDeleted]: false,
+        })
+        .timeout(Entry.DEFAULT_QUERY_TIMEOUT);
+
+    // File connections are not supported at the moment.
+    const entriesToCopy = entries.filter((entry) => {
+        const isConnection = entry.scope === EntryScope.Connection;
+        const isFile = entry.type === EntryType.File || entry.type === EntryType.GsheetsV2;
```

file connectors here should throw the error `US_ERRORS.WORKBOOK_COPY_FILE_CONNECTION_ERROR` here